v3/util/onion.go

```diff
@@ -31,24 +31,74 @@ func IsOnionV3(dnsName string) bool {
 	if len(address) != 56 {
 		return false
 	}
-	raw, err := onionV3Base32Encoding.DecodeString(address)
+	raw, err := onionBase32Encoding.DecodeString(address)
 	if err != nil {
 		return false
 	}
 	return raw[len(raw)-1] == 0x03
 }
 
-// AllAreOnionV3 returns whether-or-not EVERY name provided conforms to IsOnionV3
-func AllAreOnionV3(names []string) bool {
-	isV3 := !(len(names) == 0)
-	for _, name := range names {
-		isV3 = isV3 && IsOnionV3(name)
+// IsOnionV2Address returns whether-or-not the give address appears to be an Onion V2 address.
+//
+// In order to be an Onion V2 encoded address, the DNS name must satisfy the following:
+//	1. The address has at least two labels.
+//	2. The right most label is the .onion TLD.
+//	3. The second-to-the-right most label is a 16 character long, base32.
+func IsOnionV2Address(dnsName string) bool {
+	if !strings.HasSuffix(dnsName, "onion") {
+		return false
+	}
+	labels := strings.Split(dnsName, ".")
+	if len(labels) < 2 {
+		return false
 	}
-	return isV3
+	if len(labels[0]) != 16 {
```

Comment:
Based on the comment at the top of this function, this should check the second last label, which would be something like this:
```
address := labels[len(labels)-2]
if len(address) != 16 {
```